src/main/java/no/unit/nva/bare/AddNewAuthorityIdentifierHandler.java

```diff
@@ -1,78 +1,85 @@
 
-    public static final List<String> VALID_QUALIFIERS = asList(ValidIdentifierKey.FEIDEID.asString(),
-            ValidIdentifierKey.ORCID.asString(), ValidIdentifierKey.ORGUNITID.asString());
-
-    protected final transient BareConnection bareConnection;
-    private final transient Logger log = Logger.instance();
+    public static final List<String> VALID_QUALIFIERS = asList(
+            ValidIdentifierKey.FEIDEID.asString(),
+            ValidIdentifierKey.ORCID.asString(),
+            ValidIdentifierKey.ORGUNITID.asString());
 
+    private transient BareConnection bareConnection;
 
+    /**
+     * Default constructor for AddNewAuthorityIdentifierHandler.
+     */
+    @JacocoGenerated
     public AddNewAuthorityIdentifierHandler() {
-        this.bareConnection = new BareConnection();
-    }
-
-    public AddNewAuthorityIdentifierHandler(BareConnection bareConnection) {
-        this.bareConnection = bareConnection;
+        this(new Environment(), new BareConnection());
     }
 
     /**
-     * Main lambda function to add a qualified identifier on an existing Bare authority.
+     * Constructor for AddNewAuthorityIdentifierHandler.
      *
-     * @param input payload with path-parameters
-     * @return a GatewayResponse
+     * @param environment    environment
+     * @param bareConnection bareConnection
      */
+    public AddNewAuthorityIdentifierHandler(Environment environment, BareConnection bareConnection) {
+        super(AddNewAuthorityIdentifierRequest.class, environment,
+                LoggerFactory.getLogger(AddNewAuthorityIdentifierHandler.class));
+        this.bareConnection = bareConnection;
+    }
+
     @Override
-    @SuppressWarnings("unchecked")
-    public GatewayResponse handleRequest(final Map<String, Object> input, Context context) {
-        GatewayResponse gatewayResponse = new GatewayResponse();
+    protected Authority processInput(AddNewAuthorityIdentifierRequest input, RequestInfo requestInfo,
+                                     Context context) throws ApiGatewayException {
+
         try {
-            this.checkParameters(input);
-        } catch (RuntimeException e) {
-            log.error(e);
-            gatewayResponse.setErrorBody(e.getMessage());
-            gatewayResponse.setStatusCode(Response.Status.BAD_REQUEST.getStatusCode());
-            return gatewayResponse;
+            String scn = RequestUtils.getPathParameter(requestInfo, SCN_KEY);
+            String inputQualifier = RequestUtils.getPathParameter(requestInfo, QUALIFIER_KEY);
+            String qualifier = transformQualifier(inputQualifier);
+
+            validateInput(input);
+            String identifier = input.getIdentifier();
+            AuthorityIdentifier authorityIdentifier = new AuthorityIdentifier(qualifier, identifier);
+            return addNewIdentifier(scn, authorityIdentifier);
+        } catch (IllegalArgumentException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)